### PR TITLE
Reduce memory usage when using getFileMD5Hash function

### PR DESCRIPTION
**Pull request by rh101**
Which branch your pull-request should merge into?

- `dev`: Current 2.x BugFixs or Features

## Describe your changes
Currently, if `ax::utils::getFileMD5Hash()` is run on a file, then that entire file is loaded into memory.  If the file is very large, it results in sudden high memory usage spikes, which can cause memory issues.  To avoid this problem, I propose that we instead stream the file in smaller chunks.

The chunk size used is configurable, but is set at a default of 16KB.

## Checklist before requesting a review
-  [x] I have performed a self-review of my code.
-  [ ] If it is a core feature, I have added thorough tests.
-  [x] I have checked readme and add important infos to this PR (if it needed).
-  [ ] I have added/adapted some tests too.

### CODE REVIEW

**Comment by halx99** [comment]
Why not implement a more common API `computeFileDigest` like exist `computeDigest`, then invoke it in getFileMD5Hash

**Comment by rh101** [comment]
> Why not implement a more common API `computeFileDigest` like exist `computeDigest`, then invoke it in getFileMD5Hash

Good idea. I'll update it.

File: core/base/Utils.cpp [hunk 1]
```diff
@@ -441,46 +479,14 @@ std::string getFileMD5Hash(std::string_view filename, uint32_t bufferSize)
     {
```
Comment (halx99) [inline core/base/Utils.cpp]: don't need check totalSize, empty file also have digest

Comment (rh101) [reply core/base/Utils.cpp]: I understand. Updating it now.

**Review by halx99 — APPROVED** [review]

**Pull request merged**